R/get-path-from-object.R

```diff
@@ -158,32 +158,188 @@ get_model_id.bbi_model <- function(.mod) {
 
 #' Get path to data file
 #'
-#' Helper to extract the path to the input data from a
-#' `bbi_{.model_type}_summary` or `bbi_{.model_type}_model` object. Note that
-#' **this function only works on models that have been successfully run**
-#' because the path to the data is constructed during model submission.
-#' @param .mod Model to use, either a `bbi_{.model_type}_model` or
-#' `bbi_{.model_type}_summary` object.
+#' Helper to extract the path to the input data from a `NONMEM` control stream file.
+#'
+#' @param .bbi_object The object to query. Could be
+#' a `bbi_{.model_type}_model` object,
+#' `bbi_{.model_type}_summary` object,
+#' or a tibble of class `bbi_log_df`.
+#' @param .check_exists If `TRUE`, the default, will throw an error if the file does not exist
+#' @param pull_from_config If `TRUE`, pull the data path from `bbi.json` created
+#'  during model submission, and compare to the path defined in the control stream.
 #' @param ... Arguments passed through to methods. (Currently none, but may have
 #'   some modifier arguments other model types in the future.)
-#' @importFrom fs path_rel
+#'
+#' @details
+#'
+#' ### Notes on model extensions:
+#' `bbi` has different handling for `.mod` vs `.ctl` extensions for `NONMEM`
+#' models. When a model such as `001.ctl` is targeted with `bbi`, `bbi` will
+#' create a new directory called `001`, copy the model into it (with a `.mod`
+#' extension), and do the execution directory. For more details, see `bbi`'s
+#' [documentation](https://github.com/metrumresearchgroup/bbi/blob/main/integration/markdown/expectations.md?plain=1#L19:L21)
+#'
+#'
 #' @return Absolute path to input data file
 #' @export
-get_data_path <- function(.mod, ...) {
+get_data_path <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
   UseMethod("get_data_path")
 }
 
-#' @describeIn get_data_path Takes `bbi_{.model_type}_model` or `bbi_{.model_type}_summary` object
+#' @rdname get_data_path
 #' @export
-get_data_path.bbi_model <- function(.mod, ...) {
+get_data_path.bbi_nonmem_model <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+  get_data_path_nonmem(.bbi_object, .check_exists, pull_from_config, ...)
+}
+
+#' @rdname get_data_path
+#' @export
+get_data_path.bbi_nonmem_summary <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+  get_data_path_nonmem(.bbi_object, .check_exists, pull_from_config, ...)
+}
+
+#' @rdname get_data_path
+#' @export
+get_data_path.bbi_log_df <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+  data_paths <- map_chr(.bbi_object[[ABS_MOD_PATH]], function(.path) {
+    .mod <- read_model(.path)
+    get_data_path_nonmem(.mod, .check_exists, pull_from_config, ...)
+  })
```

Comment:
This should call the generic method, not `get_data_path_nonmem()`, because the log can consist of models that aren't `bbi_nonmem_model`.